pkg/controller/secretannotator/aws/reconciler_test.go

```diff
@@ -232,41 +243,27 @@ func mockGetUser(mockAWSClient *mockaws.MockClient) {
 }
 
 func mockSimulatePrincipalPolicyCredMinterSuccess(mockAWSClient *mockaws.MockClient) {
-	mockAWSClient.EXPECT().SimulatePrincipalPolicy(gomock.Any()).Return(&iam.SimulatePolicyResponse{
-		EvaluationResults: []*iam.EvaluationResult{
-			{EvalDecision: aws.String("allowed")},
-		},
-	}, nil)
+	mockAWSClient.EXPECT().SimulatePrincipalPolicyPages(gomock.Any(), gomock.Any()).Return(nil)
 }
 
 func mockSimulatePrincipalPolicyCredMinterFail(mockAWSClient *mockaws.MockClient) {
-	mockAWSClient.EXPECT().SimulatePrincipalPolicy(gomock.Any()).Return(&iam.SimulatePolicyResponse{
-		EvaluationResults: []*iam.EvaluationResult{
-			{
-				EvalDecision:   aws.String("notallowed"),
-				EvalActionName: aws.String("SomeAWSAction"),
-			},
-		},
-	}, nil)
+	mockAWSClient.EXPECT().SimulatePrincipalPolicyPages(gomock.Any(), gomock.Any()).Return(nil).
+		// Now in the Do() receive the lambda function f() so we can send it the failed result
```

Comment:
We should have a test case that verifies the behavior when the lambda function is supplied passing input as well.

Reply:
yes, right now it relies on the fact that an empty response means that the code assumes everything worked :/